### ISSUE TITLE
Incorrect documentation of wxDC::GetContentScaleFactor() in 3.2

### ISSUE THREAD

**Issue by taler21** [wxWidgets/wxWidgets#23469]
### Description

In master, the documentation of `wxDC::GetContentScaleFactor()` has been corrected by 5e7b08f9ea996fcb9e718b8dd0a5eb01b6d2e6e3.

This correction should also be applied to 3.2 branch.

It would probably also make sense to mention that the behavior has changed with wxWidgets-3.1.6.
Because it's an incompatible change in 3.2 to 3.0, isn't it?

I currently have a use case where I would need the previous behavior, i.e. `wxDC::GetContentScaleFactor()` behaving like `wxWindow::GetContentScaleFactor()`.
But I think I can solve the issue with the following workaround. Am I right?
```cpp
#ifdef wxHAS_DPI_INDEPENDENT_PIXELS
    const double scale = dc.GetContentScaleFactor();
#else
    const double scale = 1.0;
#endif
```

**Comment by vadz** [comment]
I'll cherry-pick the docs commit.

As for the workaround, you can indeed use it if this is what you need. I'm not sure when would it be useful though, I think you always want to use the actual scaling with `wxDC`, on all platforms...

**Comment by taler21** [comment]
Thanks for confirming that the workaround is correct.

In this case, sizes specified in 0.1 mm must be converted to logical pixels without having direct access to the window in which to draw in mapping mode `wxMM_TEXT` (i.e. only having access to the `wxDC` associated with the window).
To be DPI-aware, the conversion from 0.1 mm to logical pixels will be done in the following way:

```cpp
    // Height specified in 0.1 mm
    const int height_01mm = 35;

#ifdef wxHAS_DPI_INDEPENDENT_PIXELS
    const double scale = dc.GetContentScaleFactor();
#else
    const double scale = 1.0;
#endif
    // Resolution in logical pixels per inch
    int ppiVertical = wxRound( dc.GetPPI().GetHeight() / scale );

    // Height in logical pixels used for drawing with wxDC
    int height_px = (height_01mm * ppiVertical) / 254;
```

**Comment by vadz** [comment]
I see, thanks. Without having a window this indeed looks like the best you can do, although it is, of course, unfortunate, that a preprocessor test has to be used. But adding another function to `wxDC` is going to be even more confusing and so probably not worth it...

**Comment by taler21** [comment]
The documentation has been corrected by 8094ed579546c49c060196d1c7cfa854d8328ea3.
So this ticket can be closed. Thanks.
